Automattic/jetpack
# Stats: add new filter allowing folks to exclude IPs from tracking

Fixes #3570

#### Changes proposed in this Pull Request:

This is a first idea. It is not perfect, as per the comment in the filter's docblock. Before this is ready for a review, I think we should try the following:

- [ ] Move the `current_user_ip` method from the Jetpack class to a separate package (maybe a new class in [Status](https://github.com/Automattic/jetpack-status/tree/6300e03a808ef63dda558f0eb78e39a0e529a274/src)?).
- [ ] Move some of the tools from `projects/plugins/jetpack/modules/protect/shared-functions.php` into that same new class, so we can support not just single IPs, but IP ranges as well.

This PR does not introduce any UI. I do not think we need a UI for this extra option, since we already offer an option to exclude by role.

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Start with a site connected to WordPress.com, with the Stats feature active.
* This will require adding some code via a functionality plugin:
```php
add_filter(
	'jetpack_stats_excluded_ips',
	function ( $excluded_ips ) {
		$excluded_ips[] = 'youriphere'; // You can use something like https://www.whatsmyip.org/
		return $excluded_ips;
	}
);
```
* Access one of your posts while logged out.
* You should not see the call to `https://stats.wp.com/e-xxx` in the site's footer.
* When you remove the code snippet, you should see that call again.
```
